.github/workflows/build.yml

```diff
@@ -3,9 +3,7 @@
 
 name: Python application
 
-on:
-  push:
-    branches: [ '**' ]
+on: [push, pull_request]
```

Comment:
This will do a double build on pushes to a pull request, but this cannot be circumvented: https://github.community/t/how-to-trigger-an-action-on-push-or-pull-request-but-not-both/16662

Reply:
That's fine, our test suite doesn't take long